lib/index.js

```diff
@@ -681,22 +705,23 @@ function flushUpdate() {
  * @param {stream} stream
  * @param {*} value
  */
-function updateStreamValue(s, n) {
+function updateStreamValue(n, s) {
   s.val = n;
   s.hasVal = true;
   if (inStream === undefined) {
-    flushing = true;
-    updateDeps(s);
-    if (toUpdate.length > 0) flushUpdate(); else flushing = false;
+    flushingStreamValue = true;
+    updateListeners(s);
+    if (toUpdate.length > 0) flushUpdate();
+    flushingStreamValue = false;
   } else if (inStream === s) {
     markListeners(s, s.listeners);
   } else {
-    toUpdate.push(function() {
-      updateStreamValue(s, n);
-    });
+    updateLaterUsing(updateStreamValue_(n), s);
```

Comment:
What about using `(s) => updateStreamValue(n, s)` instead? It seems a bit simpler and we avoid using the external curry function in the implementation itself.

Reply:
Yeah, I felt like I would have to use it in more places, but didn't notice it was only in a single place. So the curry has been removed.